app/models/shipit/stack.rb

```diff
@@ -593,7 +593,13 @@ def clear_cache
 
     def update_defaults
       self.environment = 'production' if environment.blank?
-      self.branch = 'master' if branch.blank?
+      self.branch = default_branch_name if branch.blank?
+    end
+
+    def default_branch_name
+      Shipit.github.api.repo(github_repo_name).default_branch
+    rescue Octokit::NotFound, Octokit::InvalidRepository
+      'main'
```

Comment:
I think we're better to display a validation error or something like that.

Maybe just return `nil` so that the presence validator fails.

Reply:
There's no presence validation, but I can add one.